integrations/tensorflow/iree-dialects/test/iree_linalgext/tiling.mlir

```diff
@@ -159,6 +162,7 @@ func @scatter_no_tiling(
 //  CHECK-SAME:   %[[INDICES:[a-zA-Z0-9_]+]]: tensor<?x1xi32>
 //  CHECK-SAME:   %[[UPDATES:[a-zA-Z0-9_]+]]: tensor<?x?xf32>
 //       CHECK:   %[[RESULT:.+]] = iree_linalg_ext.scatter
+//  CHECK-SAME:       unique_indices(true)
```

Comment:
Instead of adding a test to check tile and distribute at Flow level, just add a test with `unique_indices` being `false` in this test file. The Flow level (after PR #8362 the TileAndDistributeToWorkgroups pass) should basically just do what is done here.

Reply:
hmm good point. I added a test but hit other issues. It will raise an error `unimplemented tiling of non-parallel loop iterator type`, and I'm not sure how to capture it in lit test.

https://github.com/google/iree/blob/c98effe27ffdcbfb81e5aa4ab6dc34abb8db8cfd/llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Tiling.cpp#L197-L205

I'll try with `-verify-diagnostics` and if it works.

Reply:
It depends on the marker you use for the test. A marker implies using a particular set of tile sizes for the test (like here https://github.com/google/iree/blob/c98effe27ffdcbfb81e5aa4ab6dc34abb8db8cfd/llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Transforms/Tiling.cpp#L339). Maybe the existing markers dont help you. EIther use some other one, or add a new one there.

Reply:
The tiling tests are mainly for "how we tile the operation given the tiling sizes". It is expected to throw out an error when the reduction dims are tiled (in scatter op). We are actually testing nothing if we don't capture the error. The testing logic would be "how do we tile and distribute the op given there are repeated indices". IMO, we should test it in `tile_and_distribute_to_workgroups.mlir`.

Reply:
you should test with a tile size of zero for that loop dimension in the case where the indices are repeated. That should compile. You can also check for it erroring out if the tile size is not zero when the indices are repeated. In any case, this is the "first line of testing". If the transformations work here, its almost certain to work when this interface is used at different places in IREE. We can chat offline about this if you still have questions.

Reply:
Got you, I added two more tests to tiling.mlir